src/script/scripting_pause_menu.cpp

```diff
@@ -60,9 +60,14 @@ bool PauseMenuScripting::checkPathInternal(const std::string &abs_path, bool wri
 	// NOTE: The pause menu env is on the same level of trust as the mainmenu env.
 	// However, since it doesn't need anything else at the moment, there's no
 	// reason to give it access to anything else.
+	// See also: `MainMenuScripting::mayModifyPath` for similar, but less restricted checks.
 
 	if (write_required)
 		return false;
-	std::string path_share = fs::AbsolutePath(porting::path_share);
-	return !path_share.empty() && fs::PathStartsWith(abs_path, path_share + DIR_DELIM "builtin");
+
+	if (write_allowed)
+		*write_allowed = false;
+
+	std::string path_builtin = fs::AbsolutePathPartial(getClient()->getBuiltinLuaPath());
+	return fs::PathStartsWith(abs_path, path_builtin);
```

Comment:
missing null check
```suggestion
	return !path_builtin.empty() && fs::PathStartsWith(abs_path, path_builtin);
```

Reply:
Can `path_builtin` ever be empty? At the very least it contains "builtin".